atlasdb-commons/src/test/java/com/palantir/common/concurrent/ConcurrentStreamsTest.java

```diff
@@ -54,30 +57,69 @@ public void testDoesEvaluateResultsWhenLimitingConcurrency() {
     }
 
     @Test
+    public void testDoesMaintainCorrectOrdering() {
+        Stream<Integer> values = ConcurrentStreams.map(
+                ImmutableList.of(5, 3, 6, 2, 1, 9),
+                value -> {
+                    if (value <= 3) {
+                        pause(100);
```

Comment:
Why should we pause here?

Reply:
Not necessary but concern was the case where the underlying data structure (i.e. used to be a map) would order based on time added. So here 6 would be processed before 3 so it should stay in the correct order after that.